resources/js/data-access/memo-api/dictionaries.ts

```diff
@@ -0,0 +1,174 @@
+import {createQuery} from "@tanstack/solid-query";
+import {LangPrefixFunc, useLangFunc} from "components/utils";
+import {translationsLoaded} from "i18n_loader";
+import {createMemo} from "solid-js";
+import {System} from "./groups";
+import {DictionaryResource, PositionResource} from "./resources/dictionary.resource";
+
+export class Dictionaries {
+  private constructor(
+    /** A map of all the dictionaries by id. */
+    readonly byId: ReadonlyMap<string, Dictionary>,
+    /**
+     * A map of the fixed dictionaries by name.
+     * Contains only the dictionaries with a translatable name (not starting with `+`).
+     */
+    readonly byName: ReadonlyMap<string, Dictionary>,
+  ) {}
+
+  static fromResources(t: LangPrefixFunc, resources: DictionaryResource[]) {
+    const byId = new Map<string, Dictionary>();
+    const byName = new Map<string, Dictionary>();
+    for (const resource of resources) {
+      const dictionary = Dictionary.fromResource(t, resource);
+      byId.set(dictionary.id, dictionary);
+      if (dictionary.resource.isFixed && dictionary.isNameTranslatable) {
+        byName.set(dictionary.resource.name, dictionary);
+      }
+    }
+    return new Dictionaries(byId, byName);
+  }
+
+  /** Returns a dictionary by id, or a fixed dictionary by name. */
+  get(idOrName: string) {
+    return this.byId.get(idOrName) || this.byName.get(idOrName);
+  }
+
+  /**
+   * Returns a subset of the dictionaries (and positions) accessible for the specified facility.
+   * If facility is not specified, dictinaries (and positions) accessible outside of facility context are returned.
```

Comment:
Personally, I would expect (based on the method name) that if the facility is undefined, it returns subset related to any facility, not unrelated to facility.

There is a separate `subsetForGlobal`, and I guess you wanted to reuse the code, but maybe both could use another private method, and this one required a defined `facilityId`?

Reply:
No dobra, zrobiłem, tylko teraz te techniczne konstruktory muszą być publiczne w rezultacie jeśli chcę sobie reużyć kod wygodnie. Jak masz jakiś elegantszy pomysł to mi daj znać.

Reply:
A może zostawić jedną metodę i zrobić tylko bardziej verbose parametr? (przyjmuję, że facilityId to uuid tak jak tam pisałeś): `subsetFor(options: {facilityId: string}|{global: true})`.

Wtedy konstruktory mogą zostać private i obie klasy mają ten sam design.

Reply:
Ustalone offline że zostawiamy jak jest.